src/marks/bollinger.js

```diff
@@ -0,0 +1,75 @@
+import {deviation, mean} from "d3";
+import {marks} from "../mark.js";
+import {map} from "../transforms/map.js";
+import {window} from "../transforms/window.js";
+import {areaX, areaY} from "./area.js";
+import {lineX, lineY} from "./line.js";
+import {identity} from "../options.js";
+
+export function bollingerBandX(
+  data,
+  {
+    x,
+    y,
+    n = 20,
+    k = 2,
+    color = "currentColor",
+    fill = color,
+    fillOpacity = 0.2,
+    stroke,
+    strokeOpacity,
+    strokeWidth,
+    ...options
+  } = {}
+) {
+  return marks(
+    areaX(data, bollingerX(n, k, {x, y1: y, y2: y, fill, fillOpacity, ...options})),
+    lineX(data, bollingerX(n, 0, {x, y, stroke, strokeOpacity, strokeWidth, ...options}))
+  );
+}
+
+export function bollingerBandY(
+  data,
+  {
+    x,
+    y,
+    n = 20,
+    k = 2,
+    color = "currentColor",
+    fill = color,
+    fillOpacity = 0.2,
+    stroke = color,
+    strokeOpacity,
+    strokeWidth,
+    ...options
+  } = {}
+) {
+  return marks(
+    areaY(data, bollingerY(n, k, {x, y1: y, y2: y, fill, fillOpacity, ...options})),
+    lineY(data, bollingerY(n, 0, {x, y, stroke, strokeOpacity, strokeWidth, ...options}))
+  );
+}
+
+export function bollingerX(n, k, options) {
```

Comment:
Not sure *n* and *k* should be required positional arguments here. Probably would be more consistent to have a *bollingerOptions* argument, and allow a single combined *options*, too. And maybe a single number could be used to specify *n*; *n* defaults to 20 and *k* defaults to 2.